flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphNotEnoughResourceTest.java

```diff
@@ -113,29 +119,28 @@ public void testRestartWithSlotSharingAndNotEnoughResources() throws Exception {
 			final JobGraph jobGraph = new JobGraph(TEST_JOB_ID, "Test Job", source, sink);
 			jobGraph.setScheduleMode(ScheduleMode.EAGER);
 
-			TestRestartStrategy restartStrategy = new TestRestartStrategy(numRestarts, false);
+			RestartBackoffTimeStrategy restartStrategy = new FixedDelayRestartBackoffTimeStrategy.FixedDelayRestartBackoffTimeStrategyFactory(numRestarts, 0).create();
 
-			final ExecutionGraph eg = TestingExecutionGraphBuilder
-				.newBuilder()
-				.setJobGraph(jobGraph)
-				.setSlotProvider(scheduler)
-				.setRestartStrategy(restartStrategy)
-				.setAllocationTimeout(Time.milliseconds(1L))
+			final SchedulerBase schedulerNG = SchedulerTestingUtils
+				.newSchedulerBuilderWithDefaultSlotAllocator(jobGraph, scheduler, Time.milliseconds(1))
+				.setRestartBackoffTimeStrategy(restartStrategy)
+				.setSchedulingStrategyFactory(new EagerSchedulingStrategy.Factory())
+				.setFailoverStrategyFactory(new RestartAllFailoverStrategy.Factory())
 				.build();
+			final ExecutionGraph eg = schedulerNG.getExecutionGraph();
```

Comment:
I agree that in the future we will make ExecutionGraph a pure topology. But I think we need to do it step by step. In my mind the first step could be factoring actions from ExecutionGraph components. And second step could be factor out all the fields unrelated to the topology. 
The first step further consists of multiple sub-steps and FLINK-15626 is just one of them. So I'd like to limit the scope of this PR to avoid make this PR too complex with multiple goals.

Regarding this specific test, you are right that it can be removed along with eager/lazy scheduling in after pipelined region scheduling is stable. So maybe let's keep the name and just remove it later?

Reply:
Now I think we can remove this test because it is actually testing job failure on slot allocation which is already covered by PipelinedRegionSchedulingITCase#testFailsOnInsufficientSlot()

Reply:
ok

Reply:
done.